pkg/vsphere/extraconfig/decode.go

```diff
@@ -0,0 +1,278 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extraconfig
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+// fromString converts string representation of a basic type to basic type
+func fromString(field reflect.Value, value string) reflect.Value {
+	switch field.Kind() {
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		s, err := strconv.ParseInt(value, 10, 64)
+		if err != nil {
+			log.Errorf("Failed to convert: %s", err.Error())
+			return field
+		}
+		return reflect.ValueOf(int(s))
+
+	case reflect.Bool:
+		s, err := strconv.ParseBool(value)
+		if err != nil {
+			log.Errorf("Failed to convert: %s", err.Error())
+			return field
+		}
+		return reflect.ValueOf(s)
+
+	case reflect.String:
+		return reflect.ValueOf(value)
+
+	case reflect.Float32, reflect.Float64:
+		s, err := strconv.ParseFloat(value, 64)
+		if err != nil {
+			log.Errorf("Failed to convert: %s", err.Error())
+			return field
+		}
+		return reflect.ValueOf(s)
+
+	}
+	log.Debugf("Invalid Kind: %s", field.Kind())
+
+	return field
+}
+
+// decodeWithPrefix convert []types.BaseOptionValue to type dest
+func decodeWithPrefix(kv map[string]string, dest interface{}, prefix string) interface{} {
+	// value representing the run-time data
+	value := reflect.ValueOf(dest).Elem()
+	log.Debugf("Value: %#v", value)
```

Comment:
I've found utility in having a global var for 'verbose' - that gets set by tests but not during standard operation. workaround for missing debug levels.